src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -1350,9 +1481,642 @@ public S(int arg)
+            }
+            else
+            {
+                Assert.Equal("System.Double C.<P>k__BackingField", fieldKeywordSymbolInfo.Symbol.ToTestDisplayString(includeNonNullable: true));
+                Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+                Assert.Same(comp.GetTypeByMetadataName("C").GetFieldsToEmit().Single(), fieldKeywordSymbolInfo.Symbol.GetSymbol());
+            }
+        }
+
+        [Theory]
+        [InlineData("get => 0; set => field = value;", false)]
+        [InlineData("set => field = value; get => 0;", false)]
+        [InlineData("get { return 0; } set { field = value; }", false)]
+        [InlineData("set { field = value; } get { return 0; }", false)]
+        [InlineData("get => 0; set { field = value; }", false)]
+        [InlineData("set { field = value; } get => 0;", false)]
+        [InlineData("get { return 0; } set => field = value;", false)]
+        [InlineData("set => field = value; get { return 0; }", false)]
+        [InlineData("get { int field = 0; return field; } set => field = value;", true)]
+        [InlineData("set => field = value; get { int field = 0; return field; }", true)]
+        public void SpeculativeSemanticModel_FieldInGetterNotUsingFieldWhereSetterAccessorDoes_BindSpeculatedFirst(string accessors, bool isLocalAfterSpeculating)
+        {
+            var comp = CreateCompilation($@"
+class C
+{{
+    public double P
+    {{
+        {accessors}
+    }}
+}}
+");
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+
+            var tree = comp.SyntaxTrees[0];
+            var token = tree.GetRoot().DescendantTokens().Single(t => t.IsKind(SyntaxKind.NumericLiteralToken));
+
+            var model = comp.GetSemanticModel(tree);
+            var identifier = SyntaxFactory.ParseExpression("field");
+            model.TryGetSpeculativeSemanticModel(token.SpanStart, (IdentifierNameSyntax)identifier, out var speculativeModel);
+            var fieldKeywordSymbolInfo = speculativeModel.GetSymbolInfo(identifier);
+            var fieldKeywordSymbolInfo2 = model.GetSpeculativeSymbolInfo(token.SpanStart, identifier, SpeculativeBindingOption.BindAsExpression);
+            Assert.Equal(fieldKeywordSymbolInfo, fieldKeywordSymbolInfo2);
+
+            Assert.Equal("System.Double C.<P>k__BackingField", comp.GetTypeByMetadataName("C").GetFieldsToEmit().Single().ToTestDisplayString());
+            if (isLocalAfterSpeculating)
+            {
+                Assert.Equal("System.Int32 field", fieldKeywordSymbolInfo.Symbol.GetSymbol().ToTestDisplayString());
+                Assert.Equal(SymbolKind.Local, fieldKeywordSymbolInfo.Symbol.Kind);
+                Assert.Equal(2, accessorBindingData.NumberOfPerformedAccessorBinding);
```

Comment:
>Assert.Equal(2, accessorBindingData.NumberOfPerformedAccessorBinding); [](http://example.com/codeflow?start=16&length=70)

I think it would be good to determine what triggers the binding (by asserting ```NumberOfPerformedAccessorBinding``` in right places), the ```GetFieldsToEmit``` above or speculative binding.